typescript/src/entities/products.ts

```diff
@@ -12,4 +16,16 @@ export class Product extends Entity {
     this._id = newId;
     this._isDirty = true;
   }
+
+  @Product.property("name")
```

Comment:
it is interesting that Typescript support such thing, is there any way that we can wrap such thing, to reduce some duplicate code?

Reply:
i investigated typescript transformers, but they are a rarely used feature. badly documented, hardly supported, and hard to use. makes the SDK harder for others to use too.

i investigated using decorators, but decorators cannot really be used to change the type (yet), so if you hack on extra properties that way, you have to add typecasts everywhere.

i thought about doing my own preprocessor, but the cure might be worse than the disease.

Reply:
i want to come back to this problem and think about it again, but, it's better to just make it work first.

for now, the sdk author have to do a bit of typing, but the user doesn't. it's not the worst thing to get started with.